liquibase/liquibase
# Mark liquibase-maven-plugin logging configuration as deprecated

## Description
In 4.0, we started ignoring the <logging> liquibase-maven-plugin configuration tag in favor of relying on the built-in maven logging system.

That means that liquibase now respects the `-q`, `-X` and `-e` flags and/or the ${maven.home}/conf/logging/simplelogger.properties file file.

But, we just silently ignored the <logging> configuration users had.

This PR outputs an ERROR level log if the <logging> configuration is set letting users know it is being ignored.

## New behavior
* If there is no <logging> configuration, there is no message
* If you have a <logging> configuration in your maven project, log a {{ The liquibase-maven-plugin now manages logging via the standard maven logging config, not the 'logging' configuration. Use the `-q`, `-X` and `-e` flags or configure Maven logging properties file following these instructions:  [https://maven.apache.org/maven](https://maven.apache.org/maven)~~logging.html}} ERROR~~level message

- - -
## Dev Handoff Notes (Internal Use)
#### Links
* Fixed Issue: Stand alone PR
* Test Results: [https://github.com/liquibase/liquibase/pull/2261/checks](https://github.com/liquibase/liquibase/pull/2261/checks)

#### Testing
* Steps to Reproduce: See above
* Guidance:
  * Impact: any liquibase-maven-plugin commands

#### Dev Verification
Verified message when and only when the logging configuration is set.

### **Test Requirements (Liquibase Internal QA)**
**Manual Tests**

_Verify that log message is displayed when with global <logging> configuration._

* specify `<liquibase.logging>debug</liquibase.logging>` in your pom.xml file under properties `section`
* execute any liquibase command
* ERROR level log message is displayed

_Verify that log message is displayed when with liquibase~~maven~~plagin <logging> configuration._

* remove global logging configuration in pom.xml file
* specify logging configuration in pom.xml file under liquibase-maven-plagin section 
```
<plugin>
		<groupId>org.liquibase</groupId>
		<artifactId>liquibase-maven-plugin</artifactId>
		<version>0-SNAPSHOT</version>
		<configuration>
			<logging>debug</logging>
		</configuration>
</plugin>
```


* execute any liquibase command
* ERROR level log message is displayed

_Verify that message is_  **_not_** _displayed when <logging> configuration is not specified._

* make sure that there is no <logging> configuration specified in pom.xml file
* execute any liquibase command
* no message is displayed

**Automated Tests**

No new functional tests required for this fix.



┆Issue is synchronized with this [Jira Bug](https://datical.atlassian.net/browse/LB-2183) by [Unito](https://www.unito.io)
